sdk/log/record.go

```diff
@@ -155,11 +273,45 @@ func (r *Record) SetAttributes(attrs ...log.KeyValue) {
 	r.back = slices.Clone(attrs[i:])
 }
 
+// head returns the first n values of kvs along with the number of elements
+// dropped. If n is less than or equal to zero, kvs is returned with 0.
+func head(kvs []log.KeyValue, n int) (out []log.KeyValue, dropped int) {
+	if n > 0 && len(kvs) > n {
+		return kvs[:n], len(kvs) - n
+	}
+	return kvs, 0
+}
+
+// dedup deduplicates kvs front-to-back with the last value saved.
+func dedup(kvs []log.KeyValue) (unique []log.KeyValue, dropped int) {
+	index := getIndex()
+	defer putIndex(index)
+
+	unique = kvs[:0] // Use the same underlying array as kvs.
+	for _, a := range kvs {
+		idx, found := index[a.Key]
```

Comment:
There can be also duplicate kvs in `index[a.Value]` if it is of `KindMap`.


See `hasDuplicatedKey` in https://github.com/pellared/opentelemetry-go/pull/10/files

Reply:
Can you link to the de-duplication requirement for values from the spec? I was not aware this was required.

Reply:
We need to apply `string` and `[]string` truncation as well. I'm planning to have it done in a follow up PR. We can address the duplicate values of map types later as well I think.

Reply:
> We can address the duplicate values of map types later as well I think.

This works for me also given that https://github.com/open-telemetry/opentelemetry-go/issues/5086 is open.

Reply:
commented.